lib/bundle.js

```diff
@@ -83,15 +87,22 @@ function calculatePath(parentFileName, referencePath) {
    * @returns {object} - Detect root files result object
    */
 function findNodeFromPath(referencePath, allData) {
+  const isReferenceRemoteURL = stringIsAValidUrl(referencePath);
+
   const partialComponents = referencePath.split(localPointer);
-  let isPartial = partialComponents.length > 1,
-    node = allData.find((node) => {
-      if (isPartial) {
-        referencePath = partialComponents[0];
-      }
-      return comparePaths(node.fileName, referencePath);
-    });
 
+  let isPartial = partialComponents.length > 1;
+  if (isPartial && !isReferenceRemoteURL) {
+    referencePath = partialComponents.slice(0, partialComponents.length - 1).join('#');
```

Comment:
@VShingala had done this refactor to test out some deep ref scenarios, but this check is just a more generic check in case we encounter reference paths that contain multiple hash fragments, but turns out we don't need this for any scenario so reverting this back to the original check